objects/src/accounts/storage/mod.rs

```diff
@@ -281,11 +290,73 @@ impl AccountStorage {
             return Err(AccountError::StorageSlotIsReserved(index));
         }
 
+        // only value slots of basic arity can currently be updated
+        match self.layout[index as usize] {
+            StorageSlotType::Value { value_arity } => {
+                if value_arity > 0 {
+                    return Err(AccountError::StorageSlotInvalidValueArity {
```

Comment:
Should the `# Errors` section of the function be update to reflect the fact only slots of arity 0 can be udpdated?

Reply:
excellent point